v2/bigquery-to-gcs/src/main/java/com/google/cloud/teleport/v2/templates/BigQueryToParquet.java

```diff
@@ -0,0 +1,291 @@
+              .setTableReference(storageTableRef);
+      try {
+        return client.createReadSession(builder.build());
+      } catch (InvalidArgumentException iae) {
+        LOG.error("Error creating ReadSession: " + iae.getMessage());
+        throw new RuntimeException(iae);
+      }
+    }
+  }
+
+  /**
+   * The {@link BigQueryToParquetOptions} class provides the custom execution options passed by the
+   * executor at the command-line.
+   */
+  public interface BigQueryToParquetOptions extends PipelineOptions {
+    @Description("BigQuery table to export from in the form <project>:<dataset>.<table>")
+    @Required
+    String getTableRef();
+
+    void setTableRef(String tableRef);
+
+    @Description("GCS bucket to export BigQuery table data to (e.g. gs://mybucket/folder/).")
+    @Required
+    String getBucket();
+
+    void setBucket(String bucket);
+
+    @Description("Optional: Number of shards for output file.")
+    @Default.Integer(0)
+    Integer getNumShards();
+
+    void setNumShards(Integer numShards);
+
+    @Description("Optional: Comma separated list of fields to select from the table.")
+    String getFields();
+
+    void setFields(String fields);
+  }
+
+  /**
+   * The {@link BigQueryToParquet#getTableSchema(ReadSession)} method gets Avro schema for table
+   * using from the {@link ReadSession} object.
+   *
+   * @param session ReadSession that contains schema for table, filtered by fields if any.
+   * @return avroSchema Avro schema for table. If fields are provided then schema will only contain
+   *     those fields.
+   */
+  private static Schema getTableSchema(ReadSession session) {
+    Schema avroSchema;
+
+    avroSchema = new Schema.Parser().parse(session.getAvroSchema().getSchema());
+    LOG.info("Schema for export is: " + avroSchema.toString());
+
+    return avroSchema;
+  }
+
+  /**
+   * Main entry point for pipeline execution.
+   *
+   * @param args Command line arguments to the pipeline.
+   */
+  public static void main(String[] args) {
+    BigQueryToParquetOptions options =
+        PipelineOptionsFactory.fromArgs(args).withValidation().as(BigQueryToParquetOptions.class);
+
+    run(options);
+  }
+
+  /**
+   * Runs the pipeline with the supplied options.
+   *
+   * @param options The execution parameters to the pipeline.
+   * @return The result of the pipeline execution.
+   */
+  private static PipelineResult run(BigQueryToParquetOptions options) {
+
+    // Create the pipeline.
+    Pipeline pipeline = Pipeline.create(options);
+
+    TableReadOptions.Builder builder = TableReadOptions.newBuilder();
+
+    /* Add fields to filter export on, if any. */
+    if (options.getFields() != null) {
+      builder.addAllSelectedFields(Arrays.asList(options.getFields().split(",\\s*")));
+    }
+
+    TableReadOptions tableReadOptions = builder.build();
+    BigQueryStorageClient client = BigQueryStorageClientFactory.create();
+    ReadSession session =
```

Comment:
Am I right that you're doing this just to fetch the schema for the read session? How could we make the BigQueryIO API surface more ergonomic here?

Reply:
@kmjung you are correct.  If there was a way to grab the schema without having to create a read session that would be ideal.